homeassistant/components/sensor/geizhals.py

```diff
@@ -46,22 +45,27 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     name = config.get(CONF_NAME)
     description = config.get(CONF_DESCRIPTION)
     product_id = config.get(CONF_PRODUCT_ID)
-    domain = config.get(CONF_DOMAIN)
-    regex = config.get(CONF_REGEX)
+    domain = config.get(CONF_LOCALE)
 
-    add_devices([Geizwatch(name, description, product_id, domain, regex)],
+    add_devices([Geizwatch(name, description, product_id, domain)],
                 True)
 
 
 class Geizwatch(Entity):
     """Implementation of Geizwatch."""
 
-    def __init__(self, name, description, product_id, domain, regex):
+    def __init__(self, name, description, product_id, domain):
         """Initialize the sensor."""
+        from geizhals import Device, Geizhals
+
+        # internal
         self._name = name
+        self._geizhals = Geizhals(product_id, domain)
+
+        # external
         self.description = description
-        self.data = GeizParser(product_id, domain, regex)
-        self._state = None
+        self.product_id = product_id
+        self.device = Device()
```

Comment:
Please use another attribute name. Entity `device` is a property now after a recent PR.